### ISSUE TITLE
new Transaction(tx.toJSON()) is incomplete

### ISSUE THREAD

**Issue by thofmann** [bitpay/bitcore#1066]
```javascript
var tx = new Transaction()
  .from(utxo, pubkeys, 2)
  .to(destinationAddress, amount)
  .fee(minerFee)
  .change(changeAddress);

console.log(tx.isFullySigned()); // false

var serialized = tx.toJSON();

var newTx = new Transaction(serialized);

console.log(newTx.isFullySigned()); // Error
```

Ideally, `tx` and `newTx` would be identical. However, each input in `newTx.toObject().inputs` does not have the following attributes that inputs in `tx.toObject().inputs` have: `threshold`, `publicKeys`, `signatures`

Without these attributes, `isFullySigned()` yields the following error:

`Unable to verify signature: Unrecognized script kind, or not enough information to execute script.This usually happens when creating a transaction from a serialized transaction`

**Comment by thofmann** [comment]
Considering it is only the inputs that differ, it is probably more of a problem with the `.toObject` method of Inputs rather than the Transaction.

**Comment by eordano** [comment]
This should be fixed in yesterday's release (0.10.2). Tried with this script and it worked all right:

```javascript
var Transaction = require('bitcore').Transaction;

var t = new Transaction()
  .from({"address":"3BazTqvkvEBcWk7J4sbgRnxUw6rjYrogf9","txid":"dc2e197ab72f71912c39bc23a42d823a3aa8d469fe65eb591c086e60d14c64a0","vout":0,"ts":1418878014,"scriptPubKey":"a9146c8d8b04c6a1e664b1ec20ec932760760c97688e87","amount":0.00300299,"confirmationsFromCache":false}, ["020483ebb834d91d494a3b649cf0e8f5c9c4fcec5f194ab94341cc99bb440007f2", "0271ebaeef1c2bf0c1a4772d1391eab03e4d96a6e9b48551ab4e4b0d2983eb452b", "03a659828aabe443e2dedabb1db5a22335c5ace5b5b7126998a288d63c99516dd8"], 2)
  .to("38nw4sTs3fCH1YiBjYeQAX1t9eWMxpek8Z", 150000)
  .change("3BazTqvkvEBcWk7J4sbgRnxUw6rjYrogf9")
  .sign("L2U9m5My3cdyN5qX1PH4B7XstGDZFWwyukdX8gj8vsJ3fkrqArQo");

var json = t.toJSON();

var f = new Transaction(json);

console.log(t.toJSON() === f.toJSON());
console.log(f.isFullySigned());

console.log(f.toJSON());
```
